### PR TITLE
mini update 3: fill in overview section

### PR DESCRIPTION
**Pull request by jdolitsky**
Fill in the overview section, including new introduction
referring to "content" (vs. just container images).

Move historical context section underneath overview.

---

This PR is built on top of #159, so the diff is misleading. Please see the following for a true diff:  https://github.com/bloodorangeio/distribution-spec/compare/mini-update-2...bloodorangeio:mini-update-3

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -5,76 +5,50 @@ keywords: registry, on-prem, images, tags, repository, distribution, api, advanc
...
 - [docker/docker-registry#612](https://github.com/docker/docker-registry/issues/612)
 
-<!--- TODO: add relevant background information here --->
+## Notational Conventions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119) (Bradner, S., "Key words for use in RFCs to Indicate Requirement Levels", BCP 14, RFC 2119, March 1997).
+
+The key words "unspecified", "undefined", and "implementation-defined" are to be interpreted as described in the [rationale for the C99 standard][c99-unspecified].
+
+An implementation is not compliant if it fails to satisfy one or more of the MUST, MUST NOT, REQUIRED, SHALL, or SHALL NOT requirements for the protocols it implements.
```
Comment (rchincha) [inline spec.md:50]: aka "minimum requirements"

Comment (jdolitsky) [reply spec.md:49]: @rchincha this is actually the old content and unrelated to this PR. Will address this in #162

**Comment by jdolitsky** [comment]
@mikebrow the updates are branched out like this: 
```
reorg
 |
 --> mini-update-1
   |
   --> mini-update-2
     |
     --> mini-update-3
```

since each of these mini updates is built on top of one another, you're just seeing the commits from unmerged updates. Once https://github.com/opencontainers/distribution-spec/pull/159 is merged, I can squash, or we can just worry about the aesthetics of this once we make a PR of reorg->master

**Comment by mikebrow** [comment]
> Once #159 is merged, I can squash, or we can just worry about the aesthetics of this once we make a PR of reorg->master

and on the twelfth day of christmas... :-)  Of course, 159 does not match the spec so once we merge it the spec becomes a spec without a matching toc... ditto for the rest of the prs.. spec-new.md? Just thinking out-loud.

**Comment by jdolitsky** [comment]
@mikebrow so very confused.. the idea is to operate on spec.md itself, and piecemeal the changes in. Instead of waiting until one PR is merged to open another, we simply opened several of them at once, built on top of one another, ordered by number (1, 2, 3 ...). The diffs are misleading, please see the PR descriptions. The TOC doesnt match the content since we havent gotten that far yet. Instead of deleting everything and starting fresh, it's important to see when/why things are being moved/deleted.

Maybe we can sync up early next week to discuss a strategy for this :)

**Comment by mikebrow** [comment]
all i'm saying is you can do it to a copy.. of spec.md

**Comment by vbatts** [comment]
LGTM

**Pull request merged**